examples/01_dirty_categories.py

```diff
@@ -1,398 +1,154 @@
-X_train.shape, X_train_enc.shape
+# Now, without much more investigation, we can already build a machine-learning
+# pipeline and train it:
 
-###############################################################################
-# Inspecting the features created
-# ...............................
-#
-# The |TV| assigns a transformer for each column. We can inspect this
-# choice:
-from pprint import pprint
+from sklearn.ensemble import HistGradientBoostingRegressor
+from sklearn.pipeline import make_pipeline
+from skrub import TableVectorizer
 
-pprint(table_vec.transformers_)
+pipeline = make_pipeline(TableVectorizer(), HistGradientBoostingRegressor())
+pipeline.fit(X, y)
 
 ###############################################################################
-# This is what is being passed to the |ColumnTransformer| under the hood.
-# If you're familiar with how the latter works, it should be very intuitive.
-# We can notice it classified the columns 'gender' and 'assignment_category'
-# as low cardinality string variables.
-# A |OneHotEncoder| will be applied to these columns.
-#
-# The vectorizer actually makes the difference between string variables
-# (data type ``object`` and ``string``) and categorical variables
-# (data type ``category``).
-#
-# Next, we can have a look at the encoded feature names.
+# What just happened there?
+# -------------------------
 #
-# Before encoding:
-X.columns.to_list()
+# First, it did not raise any errors, yay!
+# Let's explore the internals of our encoder, the |TableVectorizer|:
 
-###############################################################################
-# After encoding (we only plot the first 8 feature names):
-feature_names = table_vec.get_feature_names_out()
-feature_names[:8]
+from pprint import pprint
 
-###############################################################################
-# As we can see, it gave us interpretable columns.
-# This is because we used the |Gap| on the column 'division',
-# which was classified as a high cardinality string variable.
-# (default values, see |TV|'s docstring).
-#
-# In total, we have a reasonable number of encoded columns:
-len(feature_names)
+# Recover the TableVectorizer from the pipeline
+tv = pipeline.named_steps["tablevectorizer"]
 
+pprint(tv.transformers_)
 
 ###############################################################################
-# Feature importances in the statistical model
-# --------------------------------------------
-#
-# In this section, we will train a regressor, and plot the feature importances.
+# We observe it has automatically assigned an appropriate encoder to
+# corresponding columns.
+# For example, it classified the columns 'gender', 'department',
+# 'department_name' and 'assignment_category' as low cardinality
+# string variables.
+# Two remarkable things:, it has affected a |GapEncoder| to the columns
+# `employee_position_title` and `division`, and a |DatetimeEncoder| to the
```

Comment:
It seems that the replacement of the DatetimeEncoder did not work:
![image](https://github.com/skrub-data/skrub/assets/208217/a9948740-9748-42fe-9c41-a3c5dcf1081a)